integration-tests/e2e_tests_ci_tool/cmd/check_tests_cmd.go

```diff
@@ -0,0 +1,128 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+)
+
+type JobConfig struct {
+	Jobs map[string]struct {
+		Strategy struct {
+			Matrix struct {
+				Test []struct {
+					Path     string `yaml:"path"`
+					TestOpts string `yaml:"testOpts"`
+				} `yaml:"test"`
+			} `yaml:"matrix"`
+		} `yaml:"strategy"`
+	} `yaml:"jobs"`
+}
+
+var checkTestsCmd = &cobra.Command{
+	Use:   "check-tests [directory] [yaml file]",
+	Short: "Check if all tests in a directory are included in the test configurations YAML file",
+	Args:  cobra.ExactArgs(2),
+	Run: func(_ *cobra.Command, args []string) {
+		directory := args[0]
+		yamlFile := args[1]
+		tests, err := extractTests(directory)
+		if err != nil {
+			fmt.Println("Error extracting tests:", err)
+			os.Exit(1)
+		}
+
+		checkTestsInPipeline(yamlFile, tests)
+	},
+}
+
+func extractTests(dir string) ([]Test, error) {
+	var tests []Test
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() && strings.HasSuffix(info.Name(), "_test.go") {
+			content, err := os.ReadFile(path)
+			if err != nil {
+				return err
+			}
+			re := regexp.MustCompile(`func (Test\w+)`)
```

Comment:
my only worry is: what about tests skipped with `t.Skip()`? Seems that the tool will fail if we have skipped tests that are not part of any workflow?

also, we have scripts that build test matrices based on JSON file that's committed in the repo. So they are run, but you won't find them explicitly in any workflow. What about that use case? Will we remove these JSON files? examples are log poller or automation tests in `integration-tests` workflow

Reply:
@Tofel for skipped tests we can mark them as such in ~/.github/e2e-tests.yml